From 3809e057a13cf99fd9edca773b38bc869f9039f5 Mon Sep 17 00:00:00 2001
From: zhoulingjuan <zhoulingjuan1@huawei.com>
Date: Mon, 8 Jun 2026 09:58:38 +0800
Subject: [PATCH] umq: remove unused code (protected by external lock)
 description: remove unused code (protected by external lock)

(cherry picked from commit 85fcee405b3b8fa133d337cb39146e6cb6fff255)
---
 src/urpc/umq/qbuf/umq_shm_qbuf_pool.c | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/src/urpc/umq/qbuf/umq_shm_qbuf_pool.c b/src/urpc/umq/qbuf/umq_shm_qbuf_pool.c
index c036303..3525704 100644
--- a/src/urpc/umq/qbuf/umq_shm_qbuf_pool.c
+++ b/src/urpc/umq/qbuf/umq_shm_qbuf_pool.c
@@ -153,7 +153,6 @@ static ALWAYS_INLINE void release_thread_cache_array()
 
 static void unregister_all_thread_cache(qbuf_pool_t *pool)
 {
-    struct timespec start;
     register_list_node_t *cur = NULL;
     (void)util_rwlock_rdlock(g_register_rwlock);
     LIST_FOREACH(cur, &g_register_list_head, node) {
@@ -162,27 +161,6 @@ static void unregister_all_thread_cache(qbuf_pool_t *pool)
         if ((ref = release_thread_cache(tls_mgmt_pool)) == 0) {
             continue;
         }
-
-         /* if unregister operation release failed(which means that another thread is also operating
-          * on this thread cache), wait until the next thread finish the release operation */
-        (void)clock_gettime(CLOCK_MONOTONIC, &start);
-        uint32_t desired = ref;
-        while (!__atomic_compare_exchange_n(&tls_mgmt_pool->remove_ref_cnt, &desired, 0, true, __ATOMIC_RELEASE,
-                                            __ATOMIC_ACQUIRE)) {
-            if (desired != ref) {
-                UMQ_VLOG_ERR(VLOG_UMQ, "unexpected exception, actual ref: %u, desired ref: %u\n", ref, desired);
-                break;
-            }
-
-            if (is_timeout(&start, RELEASE_THREAD_CACHE_TIMEOUT_MS)) {
-                UMQ_VLOG_ERR(VLOG_UMQ, "release thread cache for shared memory exceeds %d ms timeout\n",
-                    RELEASE_THREAD_CACHE_TIMEOUT_MS);
-                break;
-            }
-
-            usleep(1);
-            desired = ref;
-        }
     }
     (void)util_rwlock_unlock(g_register_rwlock);
 }
-- 
2.34.1

